### DISCUSSION TITLE
Linkml configuration, dotfiles patterns, environment variables

### DISCUSSION THREAD

**Discussion by noelmcloughlin** [linkml/linkml#2296] [Ideas]
I wanted to ask opinion on whether configuration needs a long term strategy across LinkML tooling ecosystem. Typically software and tooling needs to be configurable and Linkml currently supports project level configuration like `./myproject/config.yaml` file.

Over time prolific users with many projects tend to demand global user configuration to be inherited.  The recommended implementation is typically (say)  `~/.config/linkml/generators.yaml` and  `~/.config/linkml/lint` and so on, for  [XDG](https://specifications.freedesktop.org/basedir-spec/latest/index.html#basics)  compliance, satisfying [indoctrinated believers](https://dotfiles-matter.click/) who worry about [shame trackers](https://wiki.archlinux.org/title/XDG_Base_Directory#Supported).  The project level file can override or be merged with global configuration.
    
Then there is question of environment variables.  I shared some ideas/discussion [on this project](https://github.com/linkml/linkml-project-cookiecutter/issues/57) but I need to revert a PR and rethink things.  So this is useful anchor for wider discussion, while I communicate with users on [issue](https://github.com/linkml/linkml-project-cookiecutter/issues/116).

**Comment by dalito** [comment]
See also https://github.com/linkml/linkml/pull/2335 for a proposal of a global config for LinkML.
